eland/query_compiler.py

```diff
@@ -277,18 +272,23 @@ def _es_results_to_pandas(
             # flatten row to map correctly to 2D DataFrame
             rows.append(self._flatten_dict(row, field_mapping_cache))
 
-            if batch_size is not None:
-                if i >= batch_size:
-                    partial_result = True
-                    break
+            if i % DEFAULT_PROGRESS_REPORTING_NUM_ROWS == 0:
```

Comment:
I don't think we should be yielding based on a reporting setting, maybe a batch setting instead?